core/common.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @hidden
+ */
+export interface ICommonContext {
+  /**
+   * Returns a valid SQL string that can be used to reference the dataset produced by this action.
+   */
+  self: () => string;
+
+  /**
+   * Returns the name of this dataset.
+   */
+  name: () => string;
+
+  /**
+   * References another action, returning valid SQL to be used in a `from` expression and adds it as a dependency to this action.
+   *
+   * This function can be called with a [Resolvable](#Resolvable) object, for example:
+   *
+   * ```typescript
+   * ${ref({ name: "name", schema: "schema", database: "database" })}
+   * ```
+   *
+   * This function can also be called using individual arguments for the "database", "schema", and "name" values.
+   * When only two values are provided, the default database will be used and the values will be interpreted as "schema" and "name".
+   * When only one value is provided, the default data base schema will be used, with the provided value interpreted as "name".
+   *
+   * ```typescript
+   * ${ref("database", "schema", "name")}
+   * ${ref("schema", "name")}
+   * ${ref("name")}
+   * ```
+   */
+  ref: (ref: Resolvable | string[], ...rest: string[]) => string;
+
+  /**
+   * Similar to `ref` except that it does not add a dependency, but just resolves the provided reference
+   * so that it can be used in SQL, for example in a `from` expression.
+   *
+   * See the `ref` function for example usage.
+   */
+  resolve: (ref: Resolvable | string[], ...rest: string[]) => string;
+}
+
+/**
+ * @hidden
+ */
+export interface ICommonConfig {
+  /**
+   * One or more explicit dependencies for the table action. This is typically not needed, as it is inferred from using
```

Comment:
this text is a little confusing/clumsy. suggest something more like ```One or more explicit dependencies for this action. Dependency actions will run before dependent actions. Typically this would remain unset, because most dependencies are declared as a by-product of using the `ref` function.```